src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperParseTreeRewriter.cs

```diff
@@ -324,12 +324,25 @@ private IEnumerable<string> GetAttributeNames(Block tagBlock)
                     childSpan = child as Span;
                 }
 
-                var attributeName = childSpan
-                    .Content
-                    .Split(separator: new[] { '=' }, count: 2)[0]
-                    .TrimStart();
+                var start = 0;
+                var end = 0;
+                for (; start < childSpan.Content.Length; start++)
+                {
+                    if (!char.IsWhiteSpace(childSpan.Content[start]))
+                    {
+                        break;
+                    }
+                }
+
+                for (end = start; end < childSpan.Content.Length; end++)
+                {
+                    if (childSpan.Content[end] == '=')
+                    {
+                        break;
+                    }
+                }
 
-                attributeNames.Add(attributeName);
+                attributeNames.Add(childSpan.Content.Substring(start, end - start));
```

Comment:
We could also make a bigger change here to look at the tokens individually instead of calling `.Content` and substringing. Ideally this info would be part of the syntax tree already, but that's a much bigger change.